src/components/Shifts/AddShiftForm.tsx

```diff
@@ -25,13 +25,21 @@ export interface ShiftsFormProps {
 
 export const AddShiftForm = ({port, terminal, interval, shiftForms, confirmHandler}: ShiftsFormProps) => {
   const [shifts, setShifts] = useState<ShiftForm[]>(Array.from(shiftForms).length > 0 ? shiftForms : [
-    {id: 1, name: 'Shift 1', startTime: '00:00', endTime: '01:00', defaultStaffNumber: 0}
+    {id: 1, name: '', startTime: '', endTime: '', defaultStaffNumber: 0}
   ]);
   const [showConfirm, setShowConfirm] = useState(false);
   const [error, setError] = useState(false);
+  const isFirstRender = useRef(true);
 
   useEffect(() => {
-    const hasError = shifts.some(shift => shift.name === '' || shift.startTime === shift.endTime);
+    if (isFirstRender.current) {
+      isFirstRender.current = false;
+      return;
+    }
+    const hasError = shifts.some(shift => shift.name === '' ||
```

Comment:
Doing the errors this way means errors are always displayed until the user fills in the form. Errors should only be displayed if a user has finished editing a field and it doesn't pass the validation rules.

You need to either manually track if a form field has been 'touched' or use a form helper that tracks it automatically and displays the errors when appropriate

Reply:
This is present because user can click `Add shift` without touching the shifts . I have tried with touched . It is take a more time and complex with touches .